docs/decisions/0015-expiration-improvements.rst

```diff
@@ -0,0 +1,94 @@
+This will help disambiguate behavior and simplify/remove business logic
+from the learner-portal frontend.  We may wish to also store the expiration reason
+for an assignment (i.e. which of the 3 cases above caused the assignment to become expired).
+
+Persist the allocation, cancellation, and expiration dates
+----------------------------------------------------------
+This also helps simplify and clarify the true state of an assignment for
+consumption by the learner-portal frontend.
+
+Cancellation/Expiration acknowledgement should be an action
+-----------------------------------------------------------
+There should be an action type to support this, and an endpoint scoped to the assignment identifier
+to create such action records.  This will greatly simplify logic in the learner-portal frontend
+about the state of expiration/cancellation notifications and whether they have been acknowledged/dismissed
+by the learner.
+
+Serialize the expiration deadlines with assignments
+---------------------------------------------------
+Serialize the subsidy expiration date, enrollment deadline, and 90 day deadline in the serialized
+response of assignments inside the ``credits-available`` response payload (perhaps in the
+assignments list view response payload in the future).  Additionally, return a key like ``earliest-expiration``
+that is the minimum of these three dates.
+It's ok to not have a bulk API view to acknowledge multiple assignment expiration in a single request -
+these acknowledgements are one-time activities for most assignments, and we can live with the cost
+of making multiple requests from the frontend for now.  In return, we get to deal with a simpler API view
+and logic.
+
+Modify existing expiration command
+----------------------------------
+* Compare against the allocation date of assignments to determine 90 day expiration condition.
+* Set the state to ``EXPIRED`` and set the expiration date on the expired record.
+
+Consequences
+============
+* The difference between admin-directed cancellation vs. automatic expiration becomes obvious
+  from the state of the record.
+* We store dates around the lifecycle transitions of assignment records.  This introduces a few more
+  fields to manage, but is ultimately reasonable and helpful.
+* We yank a bunch of business logic out of the frontend (which is good).
+* Serialization becomes more complex and depends on cached content metadata and subsidy metadata records.
+  We're making this tradeoff to help isolate business logic inside the backend API and model layers, which
+  is reasonable.
+* The expiration management command becomes more specific, which is good.
+
+
+Alternatives Considered
+=======================
+
+Expiration acknowledgement as an Assignment field
+-------------------------------------------------
+Rejected because it doesn't align with the pattern we've already
+introduced around assignment action records.
+
+Others
```

Comment:
Is it worth documenting the (WIP) short-term alternative from ENT-8131 here? That is, the alternative to keep the short-term approach relying on localStorage. Rejected because it allows learners to clear the dismissal history and see canceled/expired assignments again.